### PR TITLE
feat: add swiping ability for the hybrid listings page on mobile

### PR DESCRIPTION
**Pull request by ifranch**
# Pull Request Template

## Issue Overview

This PR addresses #133 

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

- Add a way to swipe the listings list up or down from the listings hybrid page. This uses the [react-swipeable](https://www.npmjs.com/package/react-swipeable) package
- Add a way to go back to the hybrid page from either the listings list only page, or the listings map only page
- Style the "swipe area"
- Create a new layout without the footer and use that layout for the listings page as well as the search-demo page. The footer will likely not be there for mobile and might just be at the end of the listings list for desktop, so it's easier to take it out now

## How Can This Be Tested/Reviewed?

- Bring up the public site and go to the listings page.
- On desktop, verify that the listings page looks the same as before: two columns, the left is the map and is fixed and the right is the listings list and is scrollable.
- Open Chrome Developer Tools and go to Toggle Device Toolbar. Pick a mobile device 
image
- You should see the listings combined page as above. Verify that you can still scroll through the listings list
- Swipe up on the region above the listings list. To do this on desktop, you should click on and drag that section up.
- You should now just see the listings list page like this 
image. Verify that you can scroll through the list and still see the swipe region at the top.
- Scroll down on the top region to go back to the combined view
- Now scroll down from the combined view, and this should take you to just the listings map page which should look like this 
image
- Verify that you can still interact with the map by clicking on a pin
- Swipe up on the bottom region to go back to the listings combined page

When testing, I am not always able to get the swiping to work. I would prefer to merge this PR and test from a mobile device using the dev page.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by isaac-looker** [comment]
scrolling the listings tab works once and then stops working. the console is emitting these seemingly relevant warnings
touch-scroll-errs

**Comment by ifranch** [comment]
> scrolling the listings tab works once and then stops working. the console is emitting these seemingly relevant warnings touch-scroll-errs

Thanks for pointing that out, I hadn't seen those warnings since they were hidden by default in Chrome. I made a small change (removing preventScrollOnSwipe: true) and that removes the warnings which makes sense based on the documentation here https://www.npmjs.com/package/react-swipeable. That doesn't fix the issue about the swiping not working the second time around, I am still looking into that but it might be some issue with react.

**Review by phattya — APPROVED** [review]
I'm also having the issue of it only consistently works the first time I either swipe up or down and then intermittently after that and sometimes not at all.

I also noticed a similar behavior the other day when testing clicking a map pin, in that the listing card would display, but then after I closed it & clicked the pin again, it wouldn't display again. I had to click another map pin & then click it again. I'm guessing this isn't related, but throwing it out there, in case it is.

I'm approving for now, so you can test on dev & see if it's any better.

**Review by chriscasto — COMMENTED** [review]
The scrolling through listings works fine for me, but swiping the list up or down only works once.  I can swipe it up then back to a neutral position or down and back to a neutral position, but after that it won't respond anymore.

Also, the listing images in the popup cards are squashed in the mobile view.  It might be better to just remove them entirely in another PR.

image

File: sites/public/src/components/listings/ListingsCombined.tsx [hunk 1]
```diff
@@ -13,20 +14,95 @@ type ListingsCombinedProps = {
...
+      setShowListingsList(true)
+      setShowListingsMap(false)
+    },
+    onSwipedDown: () => {
+      setShowListingsList(false)
+      setShowListingsMap(true)
+    },
+  })
+
+  const swipeUpHandler = useSwipeable({
```
[the remark this answers is not in the document]
Comment (ifranch) [reply sites/public/src/components/listings/ListingsCombined.tsx]: Thanks for pointing this out. I got it to work every time by having just one swipe handler and using the state variables to determine which page was swiped. Added a commit with the fix.

File: sites/public/src/components/listings/ListingsCombined.tsx [hunk 2]
```diff
@@ -13,20 +14,95 @@ type ListingsCombinedProps = {
...
+            ></ListingsList>
+          </div>
+        </div>
       </div>
-    </div>
-  )
+    )
+  }
+
+  return showListingsList
```
[the remark this answers is not in the document]
Comment (ifranch) [reply sites/public/src/components/listings/ListingsCombined.tsx]: Done, redefined it so they're both true for the listings combined page.

**Review by isaac-looker — APPROVED** [review]

**Comment by phattya** [comment]
I think this can be merged in. Chris did a fix for the build error Isaac is referring to, in a separate PR #150.

**Pull request merged**